### PR TITLE
Feature/cwall 199

### PR DESCRIPTION
**Pull request by maikel-maas**

### CODE REVIEW

File: packages/client/README.md [hunk 1]
```diff
@@ -51,6 +51,8 @@ client you can use to finish the pre-authorized code flows.
 This initiates the client using a URI obtained from the Issuer using a link (URL) or QR code typically. We are also
 already fetching the Server Metadata
 
+
+Using openid-initiate-issuance scheme
```
Comment (nklomp) [inline packages/client/README.md:55]: As mentioned that is ancient. It is called credential offer for quite some time. So if you add documentation, don't use old terms

File: packages/client/lib/CredentialOfferClient.ts [hunk 2]
```diff
@@ -27,22 +27,20 @@ export class CredentialOfferClient {
...
       credentialOffer = {
         credential_offer: credentialOfferPayload,
       };
     } else {
       credentialOffer = convertURIToJsonObject(uri, {
-        arrayTypeProperties: ['credentials'],
+        arrayTypeProperties: uri.includes('credential_offer_uri=') ? ['credential_offer_uri'] : ['credential_offer'],
         requiredProperties: uri.includes('credential_offer_uri=') ? ['credential_offer_uri'] : ['credential_offer'],
       }) as CredentialOfferV1_0_11;
-      if (credentialOffer?.credential_offer_uri === undefined && !credentialOffer?.credential_offer) {
```
Comment (nklomp) [inline packages/client/lib/CredentialOfferClient.ts:43]: Why is this removed? There should never be a by value and by reference version together in a URL

File: packages/common/lib/functions/Encoding.ts [hunk 3]
```diff
@@ -81,20 +89,47 @@ export function convertJsonToURI(
...
  *          - requiredProperties: the required properties
  *          - arrayTypeProperties: properties that can show up more that once
+ * @returns JSON object
  */
 export function convertURIToJsonObject(uri: string, opts?: DecodeURIAsJsonOpts): unknown {
   if (!uri || (opts?.requiredProperties && !opts.requiredProperties?.every((p) => uri.includes(p)))) {
     throw new Error(BAD_PARAMS);
   }
+
+  if (opts?.arrayTypeProperties && opts.arrayTypeProperties.includes('credential_offer_uri')) {
```
Comment (nklomp) [inline packages/common/lib/functions/Encoding.ts:104]: Why does a generic function all of a sudden knows about credential_offers?

File: packages/common/lib/functions/Encoding.ts [hunk 4]
```diff
@@ -81,20 +89,47 @@ export function convertJsonToURI(
...
+    return encodedCredentialOfferUri2Json({ uri })
+  }
+
+  if (opts?.arrayTypeProperties && opts.arrayTypeProperties.includes('credential_offer')) {
```
Comment (nklomp) [inline packages/common/lib/functions/Encoding.ts:108]: Why does a generic function all of a sudden know about credential_offers ?

**Review by nklomp — CHANGES_REQUESTED** [review]
See remarks

File: packages/client/README.md (hunk 1 above)
Comment (maikel-maas) [reply packages/client/README.md:55]: It was here already. Removed

File: packages/client/lib/CredentialOfferClient.ts (hunk 2 above)
Comment (maikel-maas) [reply packages/client/lib/CredentialOfferClient.ts:43]: Moved it back

Comment (nklomp) [reply packages/client/lib/CredentialOfferClient.ts:43]: Yeah, but the question is. This was a rather self-explanatory error message with a really simple condition. That raises the question, why would you even decide to remove that without consulting?

File: packages/common/lib/functions/Encoding.ts (hunk 3 above)
Comment (maikel-maas) [reply packages/common/lib/functions/Encoding.ts:104]: Removed!

File: packages/common/lib/functions/Encoding.ts (hunk 4 above)
Comment (maikel-maas) [reply packages/common/lib/functions/Encoding.ts:108]: Removed!

File: packages/common/lib/functions/Encoding.ts (hunk 3 above)
Comment (maikel-maas) [reply packages/common/lib/functions/Encoding.ts]: Adding a simple '=' to the required properties fixed the issue

File: packages/common/lib/functions/Encoding.ts (hunk 4 above)
Comment (maikel-maas) [reply packages/common/lib/functions/Encoding.ts]: Adding a simple '=' to the required properties fixed the issue

File: packages/common/lib/functions/Encoding.ts (hunk 3 above)
Comment (nklomp) [reply packages/common/lib/functions/Encoding.ts]: Hmzz, no sure I like that approach, so please create a TODO. It should probably have been solved by not mixing the outer param with the inner param

File: packages/client/lib/CredentialOfferClient.ts (hunk 2 above)
Comment (maikel-maas) [reply packages/client/lib/CredentialOfferClient.ts:43]: I didn't introduce a ternary. The ternary was already there. I just added the '=', to make it skip the scheme and split the string at the credential offer query param.

File: packages/common/lib/functions/Encoding.ts (hunk 3 above)
Comment (maikel-maas) [reply packages/common/lib/functions/Encoding.ts]: The actual problem here is the input url. Usual url encoders encode the '='. In the examples provided in the specs of OID4VCI the '=' is not encoded. If we receive a fully encoded url it doesn't work.

**Review by nklomp — APPROVED** [review]
